daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/ExpressionEvaluatingParsers.scala

```diff
@@ -165,11 +166,21 @@ class NewVariableInstanceStartParser(override val context: VariableRuntimeData)
   override lazy val runtimeDependencies = Vector()
 
   def parse(start: PState): Unit = {
-    start.newVariableInstance(context)
+    val nvi = start.newVariableInstance(context)
+
     if (context.maybeDefaultValueExpr.isDefined) {
-      val v = start.variableMap.find(context.globalQName).get
-      val res = DataValue.unsafeFromAnyRef(context.maybeDefaultValueExpr.get.evaluate(start))
-      v.setDefaultValue(res)
+      val dve = context.maybeDefaultValueExpr.get
+      dve match {
+        case constExpr: ConstantExpression[_] => // Do nothing, value has already been computed at schema compile time
```

Comment:
Remove the constant thing here so it matches unparse?

Reply:
Missed that, will fix.